src/corelibs/U2Gui/src/U2SavableWidget.h

```diff
@@ -44,17 +44,24 @@ class U2GUI_EXPORT U2SavableWidget {
     virtual void setChildValue(const QString& childId, const QVariant& value);
 
     MWMDIWindow* getContextWindow() const;
+    void appendExcludeWidgets(const QStringList& excludeWidgetIds);
 
 protected:
     virtual bool childCanBeSaved(QWidget* child) const;
     virtual QString getChildId(QWidget* child) const;
     virtual bool childExists(const QString& childId) const;
     virtual QWidget* getChildWidgetById(const QString& childId) const;
     virtual QSet<QWidget*> getCompoundChildren() const;
+    bool isExcluded(const QString& childId) const;
 
     QWidget* wrappedWidget;
     MWMDIWindow* contextWindow;
     bool widgetStateSaved;
+    /*
+     * Widgets, which state shouldn't be saved.
+     * This behaviour works for the widget and all it's children
+     */
+    QList<QWidget*> excludeWidgets;
```

Comment:
What is the widgets are dynamic (added/removed). 
It may be safier to keep ids only and check for savable widget flag only while saving

Reply:
That what I did first, but I haven't find any problems or unsafe scenarious about using pointers. It's just faster - wee call `getChildWidgetById(...)` just once for each widget instead of doing it every time this widget needed.

Reply:
`I haven't find any problems` does not mean the problem can't exist: a good code makes the problem impossible to appear.
In this case it is very easy to write a complex widget (for example with tabs) where widgets are added/removed dynamically and your solution will produce a crash.

Reply:
I think, my solution is good and have no problems. You want me to rewrite it - just say.

Reply:
Question about your solution: what if the cached widget is destroyed and replaced with another widget?
Do you still think that your solution is good?

Reply:
Ok, fixed